aws_s3_policies/aws_s3_bucket_lifecycle_configuration.py

```diff
@@ -1,15 +1,23 @@
+from panther_base_helpers import deep_get
+
 MAX_RETENTION_PERIOD = 365
 MIN_RETENTION_PERIOD = 90
 
 
 def policy(resource):
-    if resource["LifecycleRules"] is None:
+    if resource.get("LifecycleRules") is None:
         return False
 
-    for lifecycle_rule in resource["LifecycleRules"]:
-        if lifecycle_rule["Expiration"] is None or lifecycle_rule["Status"] != "Enabled":
+    for lifecycle_rule in resource.get("LifecycleRules", []):
+        if lifecycle_rule.get("Expiration") is None or lifecycle_rule.get("Status") != "Enabled":
             continue
-        if MIN_RETENTION_PERIOD <= lifecycle_rule["Expiration"]["Days"] <= MAX_RETENTION_PERIOD:
+
+        rule_retention_period_days = deep_get(lifecycle_rule, "Expiration", "Days", default=0)
+
+        if rule_retention_period_days is None:
+            continue
```

Comment:
This behavior feels inconsistent. If the `Days` key is missing, default to 0 and almost certainly fail the policy. If the `Days` key is present but mapped to `null`, then skip the lifecycle rule. Is that the intended behavior for some reason? I suspect we should not have `default=0`, allow the `default` to remain `None`, and then we're consistent at least.

Reply:
I'll just remove the new check altogether as the usage of `deep_get(...default=0)` solves the original problem

Reply:
This behavior was added to avoid a `TypeError` when doing a `<=` comparison on `None`

I'll change the conditional to be `if not rule_retention_period_days` for simplicity and remove the `default=0`